packages/valory/skills/reset_pause_abci/rounds.py

```diff
@@ -53,38 +50,6 @@ class ResetAndPauseRound(CollectSameUntilThresholdRound):
     allowed_tx_type = ResetPausePayload.transaction_type
     payload_attribute = "period_count"
 
-    def process_payload(self, payload: BaseTxPayload) -> None:  # pragma: nocover
```

Comment:
It is not identical to the base class!

The base class uses `self.synchronized_data.participants`. Here we use `self.synchronized_data.all_participants`. The difference is because we want ResetAndPauseRound to allow agents which are not currently in the set `participants` to rejoin this set.

So this "refactoring" changes the implementation.

Reply:
Solution suggestion: on the `CollectionRound` class add a flag which by default is `False` which determines whether we use `self.synchronized_data.participants` or `self.synchronized_data.all_participants`.